base_layer/core/src/base_node/comms_interface/inbound_handlers.rs

```diff
@@ -257,16 +257,29 @@ where B: BlockchainBackend + 'static
             },
             NodeCommsRequest::GetNewBlockTemplate(request) => {
                 let best_block_header = self.blockchain_db.fetch_tip_header().await?;
-                let last_seen_hash = self.mempool.get_last_seen_hash().await?;
-                let mut is_mempool_synced = true;
-                if last_seen_hash != FixedHash::default() && best_block_header.hash() != &last_seen_hash {
+                let mut last_seen_hash = self.mempool.get_last_seen_hash().await?;
+                let mut is_mempool_synced = false;
+                let mut counter = 0;
+                // this will wait a max of 100ms before returning anyway with a potential broken template
+                // We need to ensure the mempool has seen the latest base node height before we can be confident the
+                // template is correct
+                while !is_mempool_synced && counter < 10 {
```

Comment:
this could take longer than 100ms if any of the calls use a rwlock. It would be better to use a `tokio::timeout` and create a config variable with the max time to wait